src/prompt_engine/chat_engine.py

```diff
@@ -15,5 +15,24 @@ class ChatEngine(PromptEngine):
     """
     Chat Engine provides a PromptEngine to construct chat-like prompts for large scale language model inference
     """
-    def __init__(self, config: ChatEngineConfig = ChatEngineConfig(), description: str = "", examples: list = [], flow_reset_text = "", dialog: list = []):
-        super().__init__(config = config, description = description, examples = examples, flow_reset_text = flow_reset_text, dialog = dialog)
\ No newline at end of file
+    def __init__(self, config: ChatEngineConfig = ChatEngineConfig(), description: str = "", examples: list = [], flow_reset_text = "", dialog: list = [], yaml_file: str = None):
```

Comment:
I know Python has no constructor overloading, but this opens us to the case where someone passes some params _and_ a YAML representation. I guess if that happened we'd just accept the YAML config? I'm not sure what's idiomatic here.

Reply:
Ah yes, I see now that if you're passing it a yaml file the rest of the config is ignored. I think this is probably fine.